### PR TITLE
fixed: do not compare index 2 twice

### PR DESCRIPTION
**Pull request by akva2**
Found while analyzing output from static analyzers. Also sent upstream, https://github.com/OPM/ResInsight/pull/10256

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this opm-tests=963 update_data  please

File: external/resinsight/LibCore/cvfVector4.inl [hunk 1]
```diff
@@ -491,7 +491,7 @@ inline void Vector4<S>::setZero()
 template<typename S>
 inline bool Vector4<S>::isZero() const
 {
-    return (m_v[0] == 0) && (m_v[1] == 0) && (m_v[2] == 0) && (m_v[2] == 0);
+    return (m_v[0] == 0) && (m_v[1] == 0) && (m_v[2] == 0) && (m_v[3] == 0);
```
Comment (bska) [inline external/resinsight/LibCore/cvfVector4.inl:494]: Yep, that's definitely a bug&mdash;good catch!  Another option would be something like
```C++
return std::all_of(m_v, m_v + (sizeof m_v)/(sizeof m_v[0]), [](const auto& vi) { return vi == 0; });
```
which is slightly more resilient to these kinds of problems.  The downside is that that might not inline quite as well as the current version.

**Review by bska — APPROVED** [review]
Very good.  I'll merge into master.

**Comment by akva2** [comment]
jenkins build this please

**Pull request merged**